generator/views/datagroups.py

```diff
@@ -57,22 +57,35 @@ def __lt__(self, other) -> bool:
 
 
 def _get_datagroup_from_bigquery_table(
-    project_id, dataset_id, table_id, table_metadata: Dict[str, Any]
-) -> Datagroup:
+    project_id, dataset_id, table_id, table_metadata: Dict[str, Any], view: View
+) -> List[Datagroup]:
     """Use template and default values to create a Datagroup from a BQ Table."""
     full_table_id = f"{project_id}.{dataset_id}.{table_id}"
-    return Datagroup(
-        name=f"{table_id}_last_updated",
-        label=f"{table_metadata.get('friendlyName', table_id) or table_id} Last Updated",
-        description=f"Updates when {full_table_id} is modified.",
-        sql_trigger=SQL_TRIGGER_TEMPLATE.format(
-            project_id=project_id,
-            dataset_id=dataset_id,
-            table_id=table_id,
+    return [
+        Datagroup(
+            name=f"{table_id}_last_updated",
+            label=f"{table_metadata.get('friendlyName', table_id) or table_id} Last Updated",
+            description=f"Updates when {full_table_id} is modified.",
+            sql_trigger=SQL_TRIGGER_TEMPLATE.format(
+                project_id=project_id,
+                dataset_id=dataset_id,
+                table_id=table_id,
+            ),
+            # Note: Ideally we'd use the table's ETL schedule as the `maximum_cache_age` but we don't have schedule
+            # metadata here.
         ),
-        # Note: Ideally we'd use the table's ETL schedule as the `maximum_cache_age` but we don't have schedule
-        # metadata here.
-    )
+        # create a datagroup associated to a view which will be used for caching
+        Datagroup(
+            name=f"{view.name}_last_updated",
+            label=f"{view.name} Last Updated",
+            description=f"Updates for {view.name} when {full_table_id} is modified.",
+            sql_trigger=SQL_TRIGGER_TEMPLATE.format(
+                project_id=project_id,
+                dataset_id=dataset_id,
+                table_id=table_id,
+            ),
+        ),
+    ]
```

Comment:
Ideally, this should be refactored at some point since we are currently creating duplicate datagroups. However, this was necessary to associate explores with the datagroup. The datagroups that were previously generated were named based on the BigQuery table, which the explores have no context on